### PR TITLE
npm -> yarn

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

We are experiencing [large differences](https://github.com/CDCgov/prime-data-input-client/pull/299/commits/9d9c95325146a887ce726e7255086a21b650f9c4) in our `package-lock.json` due to different local npm versions. We have engines set in our [package.json](https://github.com/CDCgov/prime-data-input-client/blob/main/frontend/package.json#L5) but since engines is optional with npm the version is not aligned with our CI or cloud envs

## Changes Proposed

- add yarn
- remove npm

## Additional Information

- you will probably need to remove `.frontend/node_modules` and run `yarn`
- info on managing node version on mac https://tecadmin.net/install-nvm-macos-with-homebrew/
- yarn install info https://classic.yarnpkg.com/en/docs/install/#mac-stable

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -41,15 +41,14 @@ jobs:
...
           path: |
-            ~/.npm
             ./frontend/node_modules
-          key: npm-build-${{env.NODE_VERSION}}-${{ hashFiles('client/package-lock.json', 'client/package.json') }}
-      - name: Node setup
+          key: npm-build-${{env.NODE_VERSION}}-${{ hashFiles('frontend/yarn.lock', 'frontend/package.json') }}
+      - name: Install Dependencies
         working-directory: ./frontend
-        run: npm install
+        run: yarn
```
Comment (benwarfield-usds) [inline .github/workflows/deploy.yml:48]: I observe this is `yarn` here but `yarn install` in the next one, and I'm insufficiently confident in my yarn knowledge to know if they're equivalent or not.

File: .github/workflows/deployAzureFrontend.yml [hunk 2]
```diff
@@ -26,11 +26,11 @@ jobs:
         with:
           node-version: "14"
       - name: Install dependencies
-        run: npm install
+        run: yarn install
```
Comment (benwarfield-usds) [inline .github/workflows/deployAzureFrontend.yml:29]: Based on a consensus vote system, I think this one is wrong? 😁

File: .github/workflows/deployAzureFrontend.yml [hunk 3]
```diff
@@ -26,11 +26,11 @@ jobs:
...
       - name: Build deploy artifact
-        run: npm run build
+        run: yarn run build
```
Comment (benwarfield-usds) [inline .github/workflows/deployAzureFrontend.yml:31]: Is the `run` necessary/helpful?

**Review by benwarfield-usds — COMMENTED** [review]
Looks like there are a couple of inconsistent search-replace elements in deployAzureFrontend—not sure if they matter.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (timothybest-usds) [reply .github/workflows/deploy.yml:48]: `yarn` and `yarn install` are equivalent

File: .github/workflows/deployAzureFrontend.yml (hunk 3 above)
Comment (timothybest-usds) [reply .github/workflows/deployAzureFrontend.yml:31]: I think the `yarn` convention is not to use `run`. I think you might be able to name a script the same as a yarn command and `run` allows you to specify the script

**Pull request merged**